pkg/controller/planexecution/planexecution_controller.go

```diff
@@ -368,6 +368,10 @@ func (r *ReconcilePlanExecution) Reconcile(request reconcile.Request) (reconcile
 							if err != nil {
 								r.recorder.Event(planExecution, "Warning", "InvalidPlanExecution", fmt.Sprintf("Error expanding template: %v", err))
 								log.Printf("PlanExecutionController: Error expanding template: %v", err)
+								planExecution.Status.State = kudov1alpha1.PhaseStateError
+								planExecution.Status.Phases[i].State = kudov1alpha1.PhaseStateError
+								// returning error = nil so that we don't retry since this is non-recoverable
+								return reconcile.Result{}, nil
```

Comment:
Do we need to update the API object here?

Reply:
https://github.com/kudobuilder/kudo/blob/master/pkg/controller/planexecution/planexecution_controller.go#L252